crates/interpreter/src/instructions/memory.rs

```diff
@@ -35,3 +37,26 @@ pub fn msize(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, U256::from(interpreter.memory.effective_len()));
 }
+// From EIP-5656 MCOPY
+pub fn mcopy<SPEC: Spec>(interpreter: &mut Interpreter, _host: &mut dyn Host) {
+    check!(interpreter, SPEC::enabled(CANCUN));           
+    pop!(interpreter, dest, src, len);
+    let len = as_usize_or_fail!(interpreter, len, InstructionResult::InvalidOperandOOG);
```

Comment:
What should happen if `len==0`?
* should we just continue and skip rest of opcode.
* Should we cast dest to size.
* Should we expand memory until `dest` this would mean we could get OOG on that operation.

Reply:
I think returning early would be correct based on the geth implementation.

Reply:
Can you share the code line in geth just to compare it.

And if we just return, there is probably some base gas amount that we need to add.

Reply:
Yeah of course, I was referencing this [PR](https://github.com/ethereum/go-ethereum/pull/26181/files#diff-ab2e5d690957bc365d96e997b236ce9333d343b50dbb674d8a018aa578584cdb)